lua/ttt2/libraries/thermalvision.lua

```diff
@@ -139,7 +139,7 @@ local function RenderHook()
             local entry = thermalvisionList[i]
             local ent = entry.ent
 
-            if not IsValid(ent) then
+            if not IsValid(ent) or (ent:IsPlayer() and (ent:IsSpec() or ent:IsGhost())) then
```

Comment:
I'm not that big of a fan to include SpecDM related checks inside TTT2. Especially since this code breaks if specDM is not installed.

@mexikoedi do you have an idea how to resolve this? could this be done in SpecDM? Do we have to add a filter interface in TTT2 maybe?

Reply:
So its probably best to move the SpecDM check out of here, but the general check if the entity is a player and dead (aka a spectator) still makes sense right? (which does seem to not be used directly in TTT2 at all except for CleanUpMap)
Or if the functionality should be moved out into addons anyway removing the whole library since its not used?

Reply:
It is a bit more complicated, I get the issue now. We'd need some way to prevent ents from being rendered while they are already added. Something like a hook. Although I'm not sure if a hook is a good idea inside a render hook.

Not sure what a clean solution would be here. Maybe hide it if the player is team spec? I'd have to look a bit more into that and I'm not sure how SpecDM handles that. The same issue also exists for markerVision etc. We probably need a solution that works everywhere.

Reply:
> but the general check if the entity is a player and dead (aka a spectator) still makes sense right?

yes, that's what I'm thinking as well. Not sure how IsGhost is implemented, but if it is another team such as TEAM_TERROR and TEAM_SPEC a `not ply:IsTerror()` check should suffice.

Reply:
Updated it to use IsTerror!

Reply:
Did you test it, if it still works with SpecDM the way you intended to?

Reply:
![thermalAlive](https://github.com/user-attachments/assets/c18bb3e2-4c01-488a-995c-25f37dda76ff)
Yep works perfect! No thermal visible spectator after killing an alive player, and no visible specDM player (standing right next to alive player)

Reply:
Great that it works with `IsTerror()`.